code/game/atoms_movable.dm

```diff
@@ -21,6 +21,7 @@
 	var/mob/pulledby = null
 	var/item_state = null // Used to specify the item state for the on-mob overlays.
 	var/does_spin = TRUE // Does the atom spin when thrown (of course it does :P)
+	var/mid_diag_move = FALSE // Whether the atom is in the first step of diagonal movement
```

Comment:
This seems like a hacky way to do this. 
Have you looked at any other codebases which implement diagonal movement?

Reply:
We discussed this in Discord. To reiterate, this is necessary because diagonal movement is converted to a horizontal and vertical movement. If we want diagonal movement to actually just be a single motion, we'd need to incorporate appropriate checks for dense atoms to prevent people from moving through two objects diagonally.